yaksh/models.py

```diff
@@ -486,6 +496,33 @@ def get_user_answerpapers(self, user):
     def get_total_attempt(self, questionpaper, user):
         return self.filter(question_paper=questionpaper, user=user).count()
 
+    def get_users_for_questionpaper(self, questionpaper_id):
+        return self._get_answerpapers_for_quiz(questionpaper_id, status=True)\
+            .values("user__id", "user__first_name", "user__last_name")\
+            .distinct()
+
+    def get_user_all_attempts(self, questionpaper, user):
+        return self.filter(question_paper=questionpaper, user=user)\
+                            .order_by('-attempt_number')
+
+    def get_user_data(self, user, questionpaper_id, attempt_number = None):
+
+        if attempt_number is not None:
+            papers = self.filter( user=user, question_paper_id=questionpaper_id,
```

Comment:
please keep consistency (wherever necessary) in the space around assignment operator '=' also after comma